docs/Architecture.md

```diff
@@ -0,0 +1,502 @@
+# Babylon Staking Integration Architecture
+
+This is an architectural overview of the various components of Babylon Staking
+integration.
+
+```mermaid
+%%{init: {'theme': 'forest'}}%%
+flowchart TD
+  subgraph Babylon
+  A{{$BTC}} -- Staking --> B(Babylon staking module);
+  B -- $BTC --> C(Local Finality Provider);
+  B -- $BTC Restaking --> D(Zoneconcierge module);
+  end
+
+  D -. IBC .-> E;
+
+  subgraph BSN
+  E(Babylon Contract) -- $BTC Restake --> U(Staking contract);
+  U -- $BTC --> V(BSN Finality Provider);
```

Comment:
Why do we write "BTC" as "$BTC"? That's not terminology we use anywhere else in our docs.